flake.nix

```diff
@@ -0,0 +1,107 @@
+{
+  description = "jujitsu";
+
+  inputs.nixpkgs-mozilla.url = "github:mozilla/nixpkgs-mozilla";
+
+  outputs = { self, nixpkgs, nixpkgs-mozilla, ... }:
+    let
+      lib = nixpkgs.lib;
+      systems = [
+        "aarch64-linux"
+        "aarch64-darwin"
+        "i686-linux"
+        "x86_64-darwin"
+        "x86_64-linux"
+      ];
+      foreachSystem = f: lib.foldl' (attrs: system: lib.recursiveUpdate attrs (f system)) { } systems;
+      version = (builtins.fromTOML (builtins.readFile ./Cargo.toml)).package.version;
+    in
+    {
+      overlay = (final: prev: {
+        jujitsu = final.callPackage
+          (
+            { stdenv
+            , lib
+            , fetchFromGitHub
+            , rustPlatform
+            , pkgconfig
+            , openssl
+            , dbus
+            , sqlite
+            , file
+            , gzip
+            , makeWrapper
+            , Security
+            , SystemConfiguration
+            , libiconv
+            , rust
+            }:
+
+            rustPlatform.buildRustPackage rec {
+              pname = "jujutsu";
+              inherit version;
+
+              src = self;
+
+              cargoLock = {
+                lockFile = "${self}/Cargo.lock";
+              };
+              nativeBuildInputs = [ rust pkgconfig gzip makeWrapper ];
+              buildInputs = [ openssl dbus sqlite ]
+              ++ lib.optionals stdenv.isDarwin [
+                Security
+                SystemConfiguration
+                libiconv
+              ];
+            }
+
+          )
+          {
+            rust = (final.lib.rustLib.fromManifestFile ./toolchain-manifest.toml { inherit (final) stdenv lib fetchurl patchelf; }).rust;
+            inherit (final.darwin.apple_sdk.frameworks) Security SystemConfiguration;
+          };
+      });
+    } //
+    (foreachSystem (system:
+      let
+        pkgs = import nixpkgs {
+          inherit system;
+          overlays = [ nixpkgs-mozilla.overlays.rust self.overlay ];
+        };
+        updateToolchainManifest = pkgs.writeScriptBin "updateToolchainManifest" ''
+          #! /usr/bin/env bash
+
+          set -ex
+          
+          if [[ -z $1 ]]; then
+            channel='"nightly"'
+          else
+            channel="\"$1\""
+          fi
+
+          if [[ -z $2 ]]; then
+            date='null'
+          else
+            date="\"$2\""
+          fi
+
+          url=$(nix eval --raw --impure --expr "let flake = (builtins.getFlake (builtins.toString ./.)); in (import flake.inputs.nixpkgs { overlays = [ flake.inputs.nixpkgs-mozilla.overlays.rust ]; }).lib.rustLib.manifest_v2_url { channel = $channel; date = $date; }")
```

Comment:
Is this to support a version of Rust which isn't available in Nix? @martinvonz might be open to lowering the minimum-supported Rust version instead.

Reply:
I think nightly rust is required for this project, which isn't packaged in nixpkgs. I use the mozilla nixpkgs overlay. This requires a sha256 hash or rustup manifest file to install a rust channel hermetically. I will add the ./toolchain-manifest.toml to the gitattributes too

Reply:
I will also explain with some comments how to use the script

Reply:
I'm fine with making the project compatible with a stable Rust version if that helps. I won't have access to a computer until a week from now, so I can't make the changes myself, but I can approve PRs if anyone else wants to do it.